File: robocode2-server/src/main/java/net/robocode2/model/events/BulletHitBulletEvent.java
package net.robocode2.model.events;

import lombok.Value;
import net.robocode2.model.Bullet;

/**
 * Event sent when a bullet hits another bullet.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BulletHitBulletEvent implements Event {

	/** Bullet that hit another bullet */
	Bullet bullet;

	/** Bullet that got hit by the bullet */
	Bullet hitBullet;
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/event/BulletHitBulletEvent.kt
package dev.robocode.tankroyale.server.event

import dev.robocode.tankroyale.server.model.Bullet

/** Event sent when a bullet hits another bullet. */
data class BulletHitBulletEvent(
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bullet that hit another bullet */
    val bullet: Bullet?,

    /** Bullet that got hit by the bullet */
    val hitBullet: Bullet?,

) : Event()